DataProcessing/SmartInsiderConverter.cs

```diff
@@ -226,6 +244,105 @@ public bool Convert(DateTime date)
             return lines;
         }
 
+        /// <summary>
+        /// Processes the data to universe
+        /// </summary>
+        /// <param name="sid">security ID string</param>
+        /// <param name="ticker">ticker string</param>
+        /// <param name="data">Base class data</param>
+        /// <returns>Dictionary keyed by date yyyyMMdd that contains all the universe attribute</returns>
+        internal void ProcessUniverse(string sid, string ticker, SmartInsiderIntention data)
+        {
+            var date = $"{data.AnnouncementDate:yyyyMMdd}";
+            var cap = data.USDMarketCap;
+            var minPrice = data.MinimumPrice;
+            var maxPrice = data.MaximumPrice;
+            var amount = data.Amount;
+            var amountValue = data.AmountValue;
+            var percent = data.Percentage;
+            var dataInstance = $@"{cap},{minPrice},{maxPrice},{amount},{amountValue},{percent}";
+
+            Dictionary<string, string> dataDict;
+            if (!_intentionUniverse.TryGetValue(date, out dataDict))
+            {
+                dataDict = new Dictionary<string, string>();
+                _intentionUniverse[date] = dataDict;
+            }
+
+            string line;
+            if (!dataDict.ContainsKey(sid))
+            {
+                dataDict.Add(sid, dataInstance);
+            }
+            else
+            {
+                var oldValue = dataDict[sid].Split(",");
+
+                // Consolidate same day, same ticker value
+                var newMinPrice = Math.Min(
+                    decimal.Parse(oldValue[3], NumberStyles.Any, CultureInfo.InvariantCulture),
+                    minPrice);
+                var newMaxPrice = Math.Max(
+                    decimal.Parse(oldValue[4], NumberStyles.Any, CultureInfo.InvariantCulture),
+                    maxPrice);
+                var newAmount = int.Parse(oldValue[0]) + amount;
+                var newAmountValue = long.Parse(oldValue[1]) + amountValue;
+                var newPercent = decimal.Parse(oldValue[2], NumberStyles.Any, CultureInfo.InvariantCulture) + percent;
+
+                dataDict[sid] = $"{cap},{newMinPrice},{newMaxPrice},{newAmount},{newAmountValue},{newPercent}";
```

Comment:
Cool! Sure we can sum `amount/AmountValue/percent`?  Same for `SmartInsiderTransaction` bellow, does it happen often?

Reply:
They come in second resolution data, each slice represents a single buyback event. So, daily sum will need to sum them up.